receiver/receivercreator/factory.go

```diff
@@ -52,17 +52,22 @@ func (f *Factory) CustomUnmarshaler() component.CustomUnmarshaler {
 		}
 		c := intoCfg.(*Config)
 
-		// TODO: Change the sub-receivers to be under "receivers" to allow other config for the main receiver-creator receiver.
-		for subreceiverKey := range sourceViperSection.AllSettings() {
-			cfgSection := sourceViperSection.GetStringMap(subreceiverKey + "::config")
+		if err := sourceViperSection.Unmarshal(&c); err != nil {
+			return err
+		}
+
+		receiversCfg := viperSub(sourceViperSection, "receivers")
```

Comment:
Can we define "receivers" as a const somewhere close to `Config` and explain what it is?